parser/src/lexer/mod.rs

```diff
@@ -81,6 +81,9 @@ pub enum Token {
     #[token("-")]
     Minus,
 
+    #[token("^")]
+    Exp,
+
```

Comment:
nit: same thought about `Exp` vs `Exponent` as I left in the ast

Reply:
Went with `Exp` and changed the rest and `Number` to `Num`.